node/config/load.go

```diff
@@ -46,7 +72,51 @@ func FromReader(reader io.Reader, def interface{}) (interface{}, error) {
 	return cfg, nil
 }
 
-func ConfigUpdate(cfgCur, cfgDef interface{}, comment bool) ([]byte, error) {
+// function from raw config string to loadability
+type loadabilityCheck struct {
+	configOk       func(string) error
+	configNotFound func() error
+}
+
+func DefaultFullNodeLoadabilityCheck() loadabilityCheck {
+	configOk := func(cfgRaw string) error {
+		poisonStr := "#EnableSplitstore =" // splitstore must be set explicitly
+		if strings.Contains(cfgRaw, poisonStr) {
+			return xerrors.Errorf("Config contains disallowed string %s, refusing to load. Please explicitly set EnableSplitstore. Set it to false to keep chain management unchanged from March 2023 set it to true if you want to discard old state by default", poisonStr)
```

Comment:
> What if I just want the default?
> won't this cause every single node to fail to start on upgrade?

The point of this change (idea originating [here](https://filecoinproject.slack.com/archives/CP50PPW2X/p1678803533926769?thread_ts=1678803343.792909&cid=CP50PPW2X) is:
1) If you want the default and you have an existing, unset config: too bad.  You have to set it explicitly or your node won't start. This way users won't accidentally overwrite data.
2) If you already set your existing config explicitly then your config values won't be changing so that's ok and your node starts.
3) If you start a new node then the default will be set explicitly for you as part of this PR's change to the default config and your node will start.

> What if I comment out one EnableSplitstore line but keep another?

This is a good point I hadn't thought of.  What I would like to be doing instead is parsing the TOML and making sure that EnableSplitstore field is set there. I'll go back and take a look.

Reply:
> The point of this change (idea originating [here](https://filecoinproject.slack.com/archives/CP50PPW2X/p1678803533926769?thread_ts=1678803343.792909&cid=CP50PPW2X) is:

Can we just warn loudly and let the user fix their config if they want? If they don't, isn't that their problem? It's not like non-splitstore configs aren't supported...

Reply:
🤷 this honestly isn't something I thought of or particularly care about as a node operator myself.  Other node operators [appear to a lot](https://filecoinproject.slack.com/archives/CP50PPW2X/p1678916423822579?thread_ts=1678916327.487719&cid=CP50PPW2X) and data loss is scary so it makes some sense.  @jennijuju as the progenitor of this change could you weigh in?

Reply:
I guess going back to the original message it looks like @jennijuju is listening to the advice of glif on this one

Reply:
Supporting a safety measure they want seems reasonable

Reply:
Basically if one runs the new default on a fullnode - it will be effectively destroyed within a ~day.

Because of how expensive catch up is - restoring from backup is hardly an option.

@ZenGround0 perhaps the right direction instead of the config dance is to just check how big the badgerstore is and refuse to fuck with it over a certain size?

Reply:
Im pretty sure what actually happens is that the universal store will just stay static with all the old state ( the same way the ~130 Gigs of snapshot stay around forever) but the new state will begin to be trimmed. That's still bad but its a much easier problem to recover from. I think I'll test this and if communicate it which will hopefully lead to some more chill.

Reply:
Keeping this thread alive but the original concern is addressed in the latest commit.  We now use regex to ensure that an uncommented `EnableSplitstore =` preceded only by whitespace exists in the config which should be the correct check.

Reply:
@ribasushi I've confirmed that `datastore/chain` will stay untouched with default discard splitstore so all of the old historical data will stick around even if we don't get config dance right

Reply:
I'm suggesting the other way around. Warn loudly if the user hasn't configured anything, but leave the default the way it is for existing nodes. But I guess we'll need some form of hack either way.